src/main/java/org/elasticsearch/action/support/replication/TransportShardReplicationOperationAction.java

```diff
@@ -623,7 +783,7 @@ void performOnReplica(final ReplicationState state, final ShardRouting shard, fi
                 // to wait until they get the new mapping through the cluster
                 // state, which is why we recommend pre-defined mappings for
                 // indices using shadow replicas
-                state.onReplicaSuccess();
+                onReplicaSuccess();
```

Comment:
just an idea: we could skip the creation of the `ReplicationPhase` entirely if shadow replicas are being used.